src/runtime/wasm.ts

```diff
@@ -667,6 +679,21 @@ export class WasmParticle extends Particle {
     return p;
   }
 
+  // TODO(sjmiles): UiBroker changes ... we don't have `capabilities` for some reason, so just
+  // go straight to output
+  output(content) {
+    this.container.apiPort.Output(this, content);
+  }
+
+  // render request call-back from wasm
+  onRenderOutput(templatePtr: WasmAddress, modelPtr: WasmAddress) {
+    const content: Content = {templateName: 'default'};
+    content.template = this.container.read(templatePtr);
+    content.model = new StringDecoder().decodeDictionary(this.container.read(modelPtr));
+    this.output(content);
+  }
+
+  // TODO(sjmiles): deprecated in UiBroker contexts
   // Called by the shell to initiate rendering; the particle will call env._render in response.
```

Comment:
Can we add a JSDoc deprecated tag? 
https://jsdoc.app/tags-deprecated.html

Reply:
Ok. I'd be more proactive on this if we had an actual consumer of these annotations.

Reply:
Wouldn't it be pulled by https://live.arcs.dev/dist/apidocs/ ?

Reply:
We're going to need to do a major doc-commenting effort anyway.